.github/workflows/build-and-test.yml

```diff
@@ -5,115 +5,185 @@ on:
+  dte_agents:
+    name: DTE Agents
     runs-on: ubuntu-latest
-
-    name: Agent 1
     timeout-minutes: 20
 
+    # The Free GitHub plan has a limit of 20 concurrent jobs on Ubuntu images
+    # Reference: https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration
+    # If we need to, we can optimize for 2 simultaneous workflow runs:
+    # 2 x 1 main job   =  2 concurrent jobs
+    # 2 x 9 agent jobs = 18 concurrent jobs
+    # Total            = 20 concurrent jobs
+    #
+    # However, we don't have many projects or targets in this workspace, so we
+    # lower the number of agents to reduce spent compute time.
     strategy:
       matrix:
-        agent: [1, 2, 3]
+        agent: [1, 2, 3, 4]
 
     steps:
-      - uses: actions/checkout@v2
-      - uses: ./.github/actions/setup
+      - name: Checkout all commits
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set up dependencies
+        uses: ./.github/actions/setup
 
-      - name: Start Nx Agent ${{ matrix.agent }}
+      - name: Start Nx Cloud DTE Agent ${{ matrix.agent }}
         run: yarn nx-cloud start-agent
 
-  build:
+  # We're using Nx Cloud for Distributed Task Execution
+  # Reference: https://nx.dev/using-nx/dte
+  #
+  # The coordinator outputs the combination of task outputs from the agents,
+  # both terminal and file outputs
+  dte_coordinator:
+    name: DTE Coordinator
     runs-on: ubuntu-latest
+    env:
+      NX_CLOUD_DISTRIBUTED_EXECUTION: true
+      NX_DISTRIBUTED_TASK_EXECUTION: true
 
     steps:
-      - uses: actions/checkout@v2
-      - uses: ./.github/actions/setup
-
-      - run: yarn nx-cloud start-ci-run
-
+      - name: Checkout all commits
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set up dependencies
+        uses: ./.github/actions/setup
+
+      - name: Start Nx Cloud DTE
+        run: yarn nx-cloud start-ci-run --stop-agents-after="e2e" --agent-count=4
+
+      - name: Lint Nx workspace
+        env:
+          NX_CLOUD_DISTRIBUTED_EXECUTION: false
+          NX_DISTRIBUTED_TASK_EXECUTION: false
+        run: yarn nx workspace-lint
+      - name: Check formatting
+        env:
+          NX_CLOUD_DISTRIBUTED_EXECUTION: false
+          NX_DISTRIBUTED_TASK_EXECUTION: false
+        run: yarn nx format:check
+
+      # Distribution strategy for 2 vCPUs per hosted runner (GitHub Free):
+      #
+
+      # lint: 2 tasks assigned at a time, 1 task per vCPU
       - name: Run Affected lint
-        run: yarn nx affected:lint --parallel --maxParallel=3
+        run: yarn nx affected:lint --parallel --max-parallel=2 --quiet
 
+      # test: 1 task assigned at a time with 2 parallel processes, 1 process per vCPU
       - name: Run Affected test
-        run: yarn nx affected:test --parallel --maxParallel=3
+        id: test
+        run: yarn nx affected:test --parallel --max-parallel=1 --max-workers=2 --ci --code-coverage
```

Comment:
I don't get why max-parallel=2 when we are running 4 agents. Also, why do we need to specify workers for test and cannot use parallel execution there?

Reply:
> I don't get why max-parallel=2 when we are running 4 agents

When using Nx Cloud DTE, `--max-parallel` specifies how many tasks of this type are (at the most) distributed to **each** agent. This is what is reflected in the comment about this line. Each default hosted GitHub runner (agent) has 2 vCPUs. so it can effectively process 2 single-process tasks at the same time.

Reply:
> Also, why do we need to specify workers for test and cannot use parallel execution there?

On hosted GitHub runners with 2 vCPUs, Jest runs better when only processing one task but with `--max-workers=2`. Jest is better at distributing the load of one test project between 2 logical cores than running 2 test projects in parallel on 2 logical cores.